confirmation/models.py

```diff
@@ -108,34 +107,14 @@ def send_confirmation(self, obj, email_address, additional_context=None,
         })
         if additional_context is not None:
             context.update(additional_context)
-        if obj.realm is not None and obj.realm.is_zephyr_mirror_realm:
-            template_name = "mituser"
-        else:
-            template_name = obj._meta.model_name
-        templates = [
-            'confirmation/%s_confirmation_email.subject' % (template_name,),
-            'confirmation/confirmation_email.subject',
-        ]
-        if subject_template_path:
-            template = loader.get_template(subject_template_path)
-        else:
-            template = loader.select_template(templates)
-        subject = template.render(context).strip().replace(u'\n', u' ') # no newlines, please
-        templates = [
-            'confirmation/%s_confirmation_email.txt' % (template_name,),
-            'confirmation/confirmation_email.txt',
-        ]
-        if body_template_path:
-            template = loader.get_template(body_template_path)
-        else:
-            template = loader.select_template(templates)
-        if html_body_template_path:
-            html_template = loader.get_template(html_body_template_path)
-        else:
-            html_template = loader.get_template('confirmation/%s_confirmation_email.html' % (template_name,))
-        body = template.render(context)
-        if html_template:
-            html_content = html_template.render(context)
+
+        subject = loader.render_to_string(template_prefix + '.subject', context).strip()
+        body = loader.render_to_string(template_prefix + '.txt', context)
+        try:
+            html_content = loader.render_to_string(template_prefix + '.html', context)
+        except TemplateDoesNotExist:
+            html_content = None
```

Comment:
I'm not super excited about this try/except, since it seems like it might result in silent failures if a file has the wrong name.  Do we need this `except` case at all?  If we do, can we make it instead be a "txt_only" option we pass to this function?